src/events.rs

```diff
@@ -12,60 +10,147 @@ use std::convert::Into;
 /// adopt the same scheme as serde_json's Value: https://github.com/serde-rs/json/blob/master/json/src/value.rs
 #[derive(Clone, PartialEq)]
 pub enum Value {
-    Json(String)
+    /// Represents a JSON Boolean
+    Bool(bool),
+
+    /// Represents a JSON signed integer
+    I64(i64),
+
+    /// Represents a JSON unsigned integer
+    U64(u64),
+
+    /// Represents a JSON floating point number
+    F64(f64),
+
+    /// Represents a JSON string
+    String(String),
 }
 
 impl Value {
     // JSON serialization belongs elsewhere, but keeps
     // the distinction between regular string and JSON data clear.
-    pub fn to_json<'a>(&'a self) -> &'a str {
-        match self {
-            &Value::Json(ref s) => &s
+    pub fn to_json<'a>(&'a self) -> String {
+        match *self {
+            Value::Bool(ref b) => b.to_string(),
+            Value::I64(ref n) => n.to_string(),
+            Value::U64(ref n) => n.to_string(),
+            Value::F64(ref n) => n.to_string(),
+            Value::String(ref s) => {
+                let mut quoted = String::with_capacity(s.len() + 2);
+                quoted.push('"');
+                quoted.push_str(s);
+                quoted.push('"');
+
+                quoted
+            }
         }
     }
-    
-    /// Converts an arbitrary serializable object into the internal property format
-    /// carried on events (currently JSON values...)
-    pub fn capture<T: serde::ser::Serialize>(v: &T) -> Value {
-        Value::Json(serde_json::to_string(v).unwrap())
-    }
 }
 
 impl fmt::Debug for Value {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match self {
-            &Value::Json(ref s) => write!(f, "Value({})", s)
+        match *self {
+            Value::Bool(ref b) => write!(f, "Bool({})", b),
+            Value::I64(ref n) => write!(f, "I64({})", n),
+            Value::U64(ref n) => write!(f, "U64({})", n),
+            Value::F64(ref n) => write!(f, "F64({})", n),
+            Value::String(ref s) => write!(f, "String(\"{}\")", s)
         }
     }
 }
 
 impl fmt::Display for Value {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match self {
-            &Value::Json(ref s) => {                    
-                let bytes = s.as_bytes();            
-                if bytes.len() > 1 && bytes[0] == b'"' {
-                    write!(f, "{}", &s[1..s.len() - 1])
-                } else {
-                    write!(f, "{}", s)
-                }
-            }
+            match *self {
+            Value::Bool(ref b) => write!(f, "{}", b),
+            Value::I64(ref n) => write!(f, "{}", n),
+            Value::U64(ref n) => write!(f, "{}", n),
+            Value::F64(ref n) => write!(f, "{}", n),
+            Value::String(ref s) => write!(f, "{}", s)
         }
     }
 }
 
+impl Into<Value> for bool {
```

Comment:
I am not sure if it would implicitly overlap with anything, but could we also include an `impl<T: Into<Value>> Into<Value> for Option<T>`? This'd map to `Value::Null`.